### ISSUE TITLE
Missing response body content on 404

### ISSUE THREAD

**Issue by Rafiot** [scrapinghub/splash#1035]
This is my [lua script](https://github.com/viper-framework/ScrapySplashWrapper/blob/master/scrapysplashwrapper/crawl.lua).

Everything seems to be working fine, except that the response content of a URL returning a 404 is missing in the HAR. The content loaded from that 404 page will be there.

Is is expected? Am I missing anything?

I tried to pass `handle_httpstatus_all = True` in [my spider](https://github.com/viper-framework/ScrapySplashWrapper/blob/master/scrapysplashwrapper/crawler.py#L19), but it didn't work either for the HAR file: 

Note that if the page I request is returning a 404, `splash:html()` still give me the content.

**Comment by lucywang000** [comment]
It could be related to the target website. Have you tried some other website, with the same lua script?

**Comment by Rafiot** [comment]
Good point, I found an other website with `window.location.href` and the content is present. As we're talking of phishing websites, they are sketchy by definition. I'll try to come with an example.

For the 404, it is pretty consistent, you can try with `http://circl.lu/dsafasdfdsf` for example.

**Comment by Rafiot** [comment]
And for the empty content: https://lululujcojsta.blogspot[.]com/?m=0 (warning, phishing content)
FYI, this is the content I get with curl: [foo.txt](https://github.com/scrapinghub/splash/files/4710544/foo.txt)
